WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/AnalogWatchFaceService.kt

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.alpha
+
+import android.content.Context
+import android.graphics.RectF
+import android.graphics.drawable.Icon
+import android.util.Log
+import android.view.SurfaceHolder
+import androidx.wear.complications.ComplicationBounds
+import androidx.wear.complications.DefaultComplicationProviderPolicy
+import androidx.wear.complications.SystemProviders
+import androidx.wear.complications.data.ComplicationType
+import androidx.wear.watchface.CanvasComplicationDrawable
+import androidx.wear.watchface.CanvasType
+import androidx.wear.watchface.Complication
+import androidx.wear.watchface.ComplicationsManager
+import androidx.wear.watchface.WatchFace
+import androidx.wear.watchface.WatchFaceService
+import androidx.wear.watchface.WatchFaceType
+import androidx.wear.watchface.WatchState
+import androidx.wear.watchface.complications.rendering.ComplicationDrawable
+import androidx.wear.watchface.style.Layer
+import androidx.wear.watchface.style.UserStyleRepository
+import androidx.wear.watchface.style.UserStyleSchema
+import androidx.wear.watchface.style.UserStyleSetting
+import com.example.android.wearable.alpha.data.db.WatchFaceColorStyleEntity
+import com.example.android.wearable.alpha.viewmodels.AnalogWatchFaceViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+
+/**
+ * Handles much of the boilerplate needed to implement a watch face (minus rendering code; see
+ * [AnalogWatchCanvasRenderer]) including the complications and settings (styles user can change on
+ * the watch face).
+ */
+class AnalogWatchFaceService : WatchFaceService() {
```

Comment:
I would try to minimize the size of this Service implementation. Move out any configuration code into its own classes and according to the Single Responsibility principle this class will only be responsible for the interaction with the OS.